test/test_forestwrapper.jl

```diff
@@ -0,0 +1,29 @@
+@testset "test forestwrapper" begin
+
+    # Clean up t8code before MPI shuts down.
+    MPI.add_finalize_hook!() do
+        T8code.clean_up()
+        status = T8code.Libt8.sc_finalize_noabort()
+        # If the following test fails the allocated objects were not cleaned up
+        # properly before shutting down.
+        @test status == 0
```

Comment:
Just to double-check that I understand: Without wrapping the t8code forest in a `ForestWrapper`, this test would fail?

Reply:
Cool, thanks!